public/app/features/explore/Logs/LogsTableWrap.tsx

```diff
@@ -0,0 +1,246 @@
+import { css } from '@emotion/css';
+import { debounce } from 'lodash';
+import memoizeOne from 'memoize-one';
+import React, { useEffect, useState } from 'react';
+
+import {
+  DataFrame,
+  ExploreLogsPanelState,
+  GrafanaTheme2,
+  Labels,
+  LogsSortOrder,
+  SplitOpen,
+  TimeRange,
+} from '@grafana/data/src';
+import { Themeable2 } from '@grafana/ui/src';
+
+import { fuzzySearch } from '../../../plugins/datasource/prometheus/querybuilder/components/metrics-modal/uFuzzy';
+import { parseLogsFrame } from '../../logs/logsFrame';
+
+import { LogsColumnSearch } from './LogsColumnSearch';
+import { LogsTable } from './LogsTable';
+import { LogsTableMultiSelect } from './LogsTableMultiSelect';
+
+interface Props extends Themeable2 {
+  logsFrames: DataFrame[];
+  width: number;
+  timeZone: string;
+  splitOpen: SplitOpen;
+  range: TimeRange;
+  logsSortOrder: LogsSortOrder;
+  panelState: ExploreLogsPanelState | undefined;
+  updatePanelState: (panelState: Partial<ExploreLogsPanelState>) => void;
+  onClickFilterLabel?: (key: string, value: string, refId?: string) => void;
+  onClickFilterOutLabel?: (key: string, value: string, refId?: string) => void;
+  datasourceType?: string;
+}
+
+export type fieldNameMeta = { percentOfLinesWithLabel: number; active: boolean | undefined };
+type fieldName = string;
+type fieldNameMetaStore = Record<fieldName, fieldNameMeta>;
+
+export function LogsTableWrap(props: Props) {
+  const { logsFrames } = props;
+  // Save the normalized cardinality of each label
+  const [columnsWithMeta, setColumnsWithMeta] = useState<fieldNameMetaStore | undefined>(undefined);
+
+  // Filtered copy of columnsWithMeta that only includes matching results
+  const [filteredColumnsWithMeta, setFilteredColumnsWithMeta] = useState<fieldNameMetaStore | undefined>(undefined);
+
+  const dataFrame = logsFrames[0];
```

Comment:
Why do we show only first frame? We should probably process all frames, otherwise it can lead to confusing results, where you see at the top, that you received ~1500 logs, but table shows only 1000. 
<img width="1485" alt="image" src="https://github.com/grafana/grafana/assets/30407135/cece6f8d-4677-4cd0-8202-51216e5ec243">

Reply:
FYI: That's a limitation from my first implementation and I think that's fine to fix that in a separate PR.